conda-store-server/conda_store_server/build.py

```diff
@@ -42,9 +42,7 @@ def set_build_completed(conda_store, build, logs, packages):
 
         channel_id = api.get_conda_channel(conda_store.db, channel)
         if channel_id is None:
-            raise ValueError(
-                f"channel url={channel} not recognized in conda-store channel database"
-            )
+            channel_id = api.add_conda_channel(conda_store.db, channel)
```

Comment:
Need a conditional check `if conda_store.default_allow_channels: ...` here because ValueError should be raised if not enabled.